make build is broken with go 1.14

**Describe the bug**
make build seems to be broken with go 1.14. reverting to go 1.13.8 works fine.

```
make -C hack/tools/
make[1]: Entering directory '/home/sozercan/go/src/github.com/Azure/aks-engine/hack/tools'
make[1]: Leaving directory '/home/sozercan/go/src/github.com/Azure/aks-engine/hack/tools'
go-bindata 3.1.2 (Go runtime go1.13.8).
Copyright (c) 2010-2013, Jim Teeuwen.
go generate -mod=vendor -v ./... > /dev/null 2>&1
Makefile:83: recipe for target 'generate' failed
make: *** [generate] Error 1
```

**Steps To Reproduce**
`make build` with go 1.14

**Expected behavior**
builds binary

**AKS Engine version**
0.47.0

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/Azure/aks-engine
 
-go 1.12
+go 1.14
 
 require (
 	github.com/Azure/azure-sdk-for-go v36.2.0+incompatible

--- a/hack/tools/Makefile
+++ b/hack/tools/Makefile
@@ -17,7 +17,7 @@ $(LOCALBIN)/ginkgo:
 	GOBIN=$(LOCALBIN) $(GO) get github.com/onsi/ginkgo/ginkgo/...@v1.10.1
 
 $(LOCALBIN)/golangci-lint:
-	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(LOCALBIN) v1.21.0
+	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(LOCALBIN) v1.23.7
 
 $(LOCALBIN)/pub:
 	GOBIN=$(LOCALBIN) $(GO) get github.com/devigned/pub/...@v0.2.0

